lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -619,9 +665,11 @@ def do_3d_projection(self, renderer):
             self.update_scalarmappable()
             self._facecolors3d = self._facecolors
 
-        txs, tys, tzs = proj3d.proj_transform_vec(self._vec, renderer.M)
-        xyzlist = [(txs[si:ei], tys[si:ei], tzs[si:ei])
-                   for si, ei in self._segis]
+        xys = proj3d.proj_transform_vec(self._vec, renderer.M).T
+
+        xyzlist = _array_split(xys[:, :3],
+                               np.cumsum(self._seg_sizes),
+                               remove_empty=True)
```

Comment:
Instead of `remove_empty` here, you could just pass `np.cumsum(self._seg_sizes[:-1])`